pagopa/io-app
# feat: [IOBP-49] Discount initiatives unsubscription

## Short description
discount type initiatives now display the correct amount of checks upon entering the unsubscription flow

## List of changes proposed in this pull request
- added a new redux selector to pass the initiative type to the unsub machine
- refactored the machine to accept said data
- added a selector in the machine to provide the initiative's type
- fixed tests where needed
the unsubscription flow is still startable from any point of the app, and will fetch the required data if started with only an initiative ID.

## How to test
navigate to an initiative's detail and start the unsubscription flow. make sure that in both refund and discount initiatives the correct checks are displayed as per design.
